tests/optim/test_layerwise_gradient_scaler.py

```diff
@@ -19,6 +19,7 @@
 
 from fairscale.optim.layerwise_gradient_scaler import LayerwiseGradientScaler
 from fairscale.utils.testing import skip_a_test_if_in_CI
+from tests.common_paths import DATASET_CACHE_ROOT
```

Comment:
See my comment from the not-a-pr PR. I suggest we don't import code from `tests` dir since (1) that top level module name is very common and keeping all code under fairscale make things easier to grep etc. How about adding this to fairscale/testing dir?

Reply:
this is a deeper question, about what we want to export to library users, vrs the amount of machinery we want to include in the testing stack (and the development package requirements vrs the user package requirements).

as a general pattern, the current tests don't use project-specific testing libraries, so the robustness of those libs, when dropped into `fairscale/testing` is limited to what we wish to expose to experimental users; which has a cooling effect on them.

another path, `fairscale_testlibs/` could exist, to hold this machinery, and distinguish it from `fairscale/` and `tests/`

I would like to add that this pr (make `.gitignore` play nice with current behavior) is growing in scope to (fix common test data paths as a pattern); but I'm happy to finish this.

as a concrete proposal, I suggest `fair_testlibs/` as a path, and if we later extract that to a supplemental dev repo, we don't have to change the module paths

Reply:
Your plan sounds great to me

Reply:
Updated, but used `fair_dev/` to cover other potential dev-only machinery.